### PR TITLE
fix: update for API route change

### PR DESCRIPTION
**Pull request by fubuloubu**
### What I did

just changed the API route

fixes: #46

### Checklist

- [x] Passes all linting checks (pre-commit and CI jobs)
- [x] New test cases have been added and are passing
- [x] Documentation has been updated
- [x] PR title follows [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) standard (will be automatically included in the changelog)

### CODE REVIEW

**Comment by fubuloubu** [comment]
TODO:
- [ ] Check all endpoints for changes

**Comment by solarthesis** [comment]
Hey @fubuloubu  I believe we still have issues, here is what it is throwing
```
*** pydantic_core._pydantic_core.ValidationError: 13 validation errors for UnexecutedTxData
safe
  Field required [type=missing, input_value={'executionDate': '2024-0...934F4F0C45f893042B190b'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.7/v/missing
value
  Field required [type=missing, input_value={'executionDate': '2024-0...934F4F0C45f893042B190b'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.7/v/missing
operation
  Field required [type=missing, input_value={'executionDate': '2024-0...934F4F0C45f893042B190b'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.7/v/missing
gasToken
  Field required [type=missing, input_value={'executionDate': '2024-0...934F4F0C45f893042B190b'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.7/v/missing
safeTxGas
  Field required [type=missing, input_value={'executionDate': '2024-0...934F4F0C45f893042B190b'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.7/v/missing
baseGas
  Field required [type=missing, input_value={'executionDate': '2024-0...934F4F0C45f893042B190b'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.7/v/missing
gasPrice
  Field required [type=missing, input_value={'executionDate': '2024-0...934F4F0C45f893042B190b'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.7/v/missing
refundReceiver
  Field required [type=missing, input_value={'executionDate': '2024-0...934F4F0C45f893042B190b'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.7/v/missing
nonce
  Field required [type=missing, input_value={'executionDate': '2024-0...934F4F0C45f893042B190b'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.7/v/missing
submissionDate
  Field required [type=missing, input_value={'executionDate': '2024-0...934F4F0C45f893042B190b'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.7/v/missing
modified
  Field required [type=missing, input_value={'executionDate': '2024-0...934F4F0C45f893042B190b'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.7/v/missing
safeTxHash
  Field required [type=missing, input_value={'executionDate': '2024-0...934F4F0C45f893042B190b'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.7/v/missing
confirmationsRequired
  Field required [type=missing, input_value={'executionDate': '2024-0...934F4F0C45f893042B190b'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.7/v/missing
    ````
    
    Here is an example I am trying to push
    ```
    from ape import Contract, networks
from ape.exceptions import SignatureError as SigError
from ape_safe import multisend
from functools import wraps
from ape import accounts

class GnosisSafe():
    
    def __init__(self):
        self.setup_deployer()
        self.setup_safe()

    def with_provider(func):
        @wraps(func)
        def wrapper(self,*args, **kwargs):
            provider  = getattr(getattr(networks,"optimism"),"mainnet")
            with provider.use_default_provider():
                return func(self,*args, **kwargs)
        return wrapper
        
    @with_provider
    def setup_safe(self):
        self.safe = accounts.load("test-safe")
    
    def setup_deployer(self):  
        self.deployer = accounts.load("test")
        self.deployer.set_autosign(True,passphrase='abc123')
                
    
    @with_provider
    def get_signers(self):
        return self.safe.signers
    
    @with_provider
    def get_next_safe_nonce(self):
        return self.safe.new_nonce

    @with_provider
    def post_gnosis_transaction(self,contractName,functionName,args):
        nonce          = self.get_next_safe_nonce()
        apeContract    = self.apeContracts[contractName]
        func           = getattr(apeContract,functionName)
        txn            = multisend.MultiSend()
        [txn.add(func,*arg) for arg in args]
        try:        
            txn(sender=self.safe,gas=0,submit_transaction=False,nonce=nonce)
        except SigError:                
            pass
        except Exception as e: 
            print(f"error staging transaction {e}")
            
            
    @with_provider
    def stage_test_transaction(self):
        abi='''[{"constant":false,"inputs":[{"name":"dst","type":"address"},{"name":"wad","type":"uint256"}],"name":"transfer","outputs":[{"name":"","type":"bool"}],"payable":false,"stateMutability":"nonpayable","type":"function"}]'''
        self.apeContracts = {}
        self.apeContracts["test"] = Contract(address="0x4200000000000000000000000000000000000006",abi=abi)
        self.post_gnosis_transaction(contractName='test', functionName='transfer', args=[[self.safe.address,1]]) 
    
#%%
if __name__ == '__main__':
    self = GnosisSafe()
    self.stage_test_transaction()
    ```
 
 And here is the versions installed
 ```
 (gnosis_perp_ape) thesis@ks:~$ ape --version
0.8.2

 Installed Plugins
  alchemy      0.8.0
  etherscan    0.8.0
  solidity     0.8.0

Third-party Plugins
  optimism     0.8.0
  safe         0.8.1.dev1+g6ff9246

```

**Comment by fubuloubu** [comment]
@solarthesis confirmed this locally just using `ape safe all-txns`

**Comment by solarthesis** [comment]
@fubuloubu believe there are still things i'ts not liking
```
(gnosis_perp_ape) thesis@ks:~$ ape safe all-txns --network optimism:mainnet 0x2008A7DDf4770840B9C301A2722564b07ec7800c
INFO: Connecting to existing Geth node at https://mainnet.optimism.io/[hidden].
Traceback (most recent call last):
  File "/home/thesis/miniconda3/envs/gnosis_perp_ape/bin/ape", line 8, in <module>
    sys.exit(cli())
             ^^^^^
  File "/home/thesis/miniconda3/envs/gnosis_perp_ape/lib/python3.11/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/thesis/miniconda3/envs/gnosis_perp_ape/lib/python3.11/site-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/home/thesis/miniconda3/envs/gnosis_perp_ape/lib/python3.11/site-packages/ape/_cli.py", line 88, in invoke
    return super().invoke(ctx)
           ^^^^^^^^^^^^^^^^^^^
  File "/home/thesis/miniconda3/envs/gnosis_perp_ape/lib/python3.11/site-packages/click/core.py", line 1688, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/thesis/miniconda3/envs/gnosis_perp_ape/lib/python3.11/site-packages/click/core.py", line 1688, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/thesis/miniconda3/envs/gnosis_perp_ape/lib/python3.11/site-packages/ape/cli/commands.py", line 98, in invoke
    return self._invoke(ctx, provider=provider)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/thesis/miniconda3/envs/gnosis_perp_ape/lib/python3.11/site-packages/ape/cli/commands.py", line 135, in _invoke
    return ctx.invoke(self.callback or (lambda: None), **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/thesis/miniconda3/envs/gnosis_perp_ape/lib/python3.11/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/thesis/miniconda3/envs/gnosis_perp_ape/lib/python3.11/site-packages/click/decorators.py", line 92, in new_func
    return ctx.invoke(f, obj, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/thesis/miniconda3/envs/gnosis_perp_ape/lib/python3.11/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/thesis/miniconda3/envs/gnosis_perp_ape/lib/python3.11/site-packages/ape_safe/_cli/safe_mgmt.py", line 145, in all_txns
    for txn in client.get_transactions(confirmed=confirmed):
  File "/home/thesis/miniconda3/envs/gnosis_perp_ape/lib/python3.11/site-packages/ape_safe/client/base.py", line 82, in get_transactions
    for txn in self._all_transactions():
  File "/home/thesis/miniconda3/envs/gnosis_perp_ape/lib/python3.11/site-packages/ape_safe/client/__init__.py", line 95, in _all_transactions
    yield UnexecutedTxData.model_validate(txn)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/thesis/miniconda3/envs/gnosis_perp_ape/lib/python3.11/site-packages/pydantic/main.py", line 551, in model_validate
    return cls.__pydantic_validator__.validate_python(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
pydantic_core._pydantic_core.ValidationError: 13 validation errors for UnexecutedTxData
safe
  Field required [type=missing, input_value={'executionDate': '2024-0...934F4F0C45f893042B190b'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.7/v/missing
value
  Field required [type=missing, input_value={'executionDate': '2024-0...934F4F0C45f893042B190b'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.7/v/missing
operation
  Field required [type=missing, input_value={'executionDate': '2024-0...934F4F0C45f893042B190b'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.7/v/missing
gasToken
  Field required [type=missing, input_value={'executionDate': '2024-0...934F4F0C45f893042B190b'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.7/v/missing
safeTxGas
  Field required [type=missing, input_value={'executionDate': '2024-0...934F4F0C45f893042B190b'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.7/v/missing
baseGas
  Field required [type=missing, input_value={'executionDate': '2024-0...934F4F0C45f893042B190b'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.7/v/missing
gasPrice
  Field required [type=missing, input_value={'executionDate': '2024-0...934F4F0C45f893042B190b'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.7/v/missing
refundReceiver
  Field required [type=missing, input_value={'executionDate': '2024-0...934F4F0C45f893042B190b'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.7/v/missing
nonce
  Field required [type=missing, input_value={'executionDate': '2024-0...934F4F0C45f893042B190b'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.7/v/missing
submissionDate
  Field required [type=missing, input_value={'executionDate': '2024-0...934F4F0C45f893042B190b'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.7/v/missing
modified
  Field required [type=missing, input_value={'executionDate': '2024-0...934F4F0C45f893042B190b'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.7/v/missing
safeTxHash
  Field required [type=missing, input_value={'executionDate': '2024-0...934F4F0C45f893042B190b'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.7/v/missing
confirmationsRequired
  Field required [type=missing, input_value={'executionDate': '2024-0...934F4F0C45f893042B190b'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.7/v/missing
(gnosis_perp_ape) thesis@ks:~$ 


```

**Comment by fubuloubu** [comment]
Yeah was just saying I got the same issue that way haha, working on debugging

**Comment by fubuloubu** [comment]
@solarthesis should be fixed now!

**Comment by solarthesis** [comment]
ok let me test it out, 1 second

**Comment by solarthesis** [comment]
Works perfectly @fubuloubu 
image

**Review by antazoey — APPROVED** [review]
couple o' nits / small-refactor opportunities

**Comment by solarthesis** [comment]
@Uxio0 please take note of breaking changes on gnosis api, which is causing issues for integrations like this  https://github.com/safe-global/safe-transaction-service/commit/6753c7aa747d18f6fb86849a6007a4b7fa7f4fe8

File: ape_safe/client/__init__.py [hunk 1]
```diff
@@ -77,22 +77,24 @@ def get_next_nonce(self) -> int:
...
         while url:
             response = self._get(url)
             data = response.json()
 
             for txn in data.get("results"):
-                # TODO: Replace with `model_validate()` after ape 0.7.
                 # NOTE: Using construct because of pydantic v2 back import validation error.
-                if "isExecuted" in txn and txn["isExecuted"]:
-                    yield ExecutedTxData.model_validate(txn)
+                if "isExecuted" in txn:
```
[the remark this answers is not in the document]
Comment (fubuloubu) [reply ape_safe/client/__init__.py:90]: There's actually 3 logical conditions here, None True or False

I need to skip the None case entirely (incoming txn),  True/False dictates if it has been confirmed

File: ape_safe/_cli/safe_mgmt.py [hunk 2]
```diff
@@ -132,12 +132,16 @@ def remove(cli_ctx: SafeCliContext, safe, skip_confirmation):
...
 @safe_cli_ctx()
-@click.argument("address", type=ChecksumAddress)
+@click.argument("account")
 @click.option("--confirmed", is_flag=True, default=None)
-def all_txns(cli_ctx: SafeCliContext, address, confirmed):
+def all_txns(cli_ctx: SafeCliContext, account, confirmed):
     """
     View and filter all transactions for a given Safe using Safe API
     """
+    if account in cli_ctx.account_manager.aliases:
```
[the remark this answers is not in the document]
Comment (fubuloubu) [reply ape_safe/_cli/safe_mgmt.py:141]: Yes but I think could be upstreamed to ape.cli

File: ape_safe/client/__init__.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (antazoey) [reply ape_safe/client/__init__.py:90]: ahh i read this wrong!

**Pull request merged**